### PR TITLE
Shallow-clone git step references

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-io/bitrise/blob/master/.github/CONTRIBUTING.md)
- [ ] `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

When loading steps from git repos instead of the steplib, we do a full git clone at the moment. This is a bit wasteful because we only need the latest state of the repo, not its entire history.

### Changes

Perform a shallow clone whenever the CLI clones a repo.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.16
 
 require (
 	github.com/bitrise-io/envman v0.0.0-20221010094751-a03ce30a5316
-	github.com/bitrise-io/go-utils v1.0.3
+	github.com/bitrise-io/go-utils v1.0.4-0.20221229083519-3e15e6ea1f2d
```
Comment (hisaac) [inline go.mod:7]: Any idea why this doesn't just resolve to `v1.0.4`? Maybe it doesn't matter ¯\\\_(ツ)_/¯, just weird that it wouldn't resolve to the tag, unless you did it on purpose?

**Review by hisaac — COMMENTED** [review]
Seems good to me, but I see this is still in draft, so I'll wait to officially approve.

File: go.mod (hunk 1 above)
Comment (hisaac) [reply go.mod:7]: oh nevermind, I get it now. You're pulling in the changes from https://github.com/bitrise-io/go-utils/pull/176 here. Carry on!

Comment (ofalvai) [reply go.mod]: Yup, this is why I kept this PR as a draft! Now that I merged the go-utils PR and tagged it as 1.0.4, I pushed a commit that fixes this.

**Review by hisaac — APPROVED** [review]

**Pull request merged**